modules/internal/Bytes.chpl

```diff
@@ -83,6 +83,166 @@ module Bytes {
     }
   }
 
+  //
+  // createBytes* functions
+  //
+
+  /*
+    Creates a new bytes which borrows the internal buffer of another bytes. If
+    the buffer is freed before the bytes returned from this function, accessing
+    it is undefined behavior.
+
+    :arg s: Object to borrow the buffer from
+    :type s: `bytes`
+
+    :returns: A new `bytes`
+  */
+  inline proc createBytesWithBorrowedBuffer(s: bytes) {
+    var ret: bytes;
+    initWithBorrowedBuffer(ret, s);
+    return ret;
+  }
+
+  /*
+    Creates a new bytes which borrows the internal buffer of a `c_string`. If
+    the buffer is freed before the bytes returned from this function, accessing
+    it is undefined behavior.
+
+    :arg s: Object to borrow the buffer from
+    :type s: c_string
+
+    :arg length: Length of the `c_string` in bytes, excluding the terminating
+                 null byte.
+    :type length: int
+
+    :returns: A new `bytes`
+  */
+  inline proc createBytesWithBorrowedBuffer(s: c_string, length=s.length) {
+    return createBytesWithBorrowedBuffer(s:c_ptr(uint(8)), length=length,
+                                                            size=length+1);
+  }
+
+  /*
+     Creates a new bytes which borrows the memory allocated for a
+     `c_ptr(uint(8))`. If the buffer is freed before the bytes returned from
+     this function, accessing it is undefined behavior.
+
+     :arg s: Object to borrow the buffer from
+     :type s: `bufferType` (i.e. `c_ptr(uint(8))`)
+
+     :arg length: Length of the bytes stored in `s`, excluding the terminating
+                  null byte.
+     :type length: int
+
+     :arg size: Size of memory allocated for `s` in bytes
+     :type length: int
+
+     :returns: A new `bytes`
+  */
+  inline proc createBytesWithBorrowedBuffer(s: bufferType, length: int, size: int) {
+    var ret: bytes;
+    initWithBorrowedBuffer(ret, s, length,size);
+    return ret;
+  }
+
+  pragma "no doc"
+  inline proc createBytesWithOwnedBuffer(s: bytes) {
+    // should we allow stealing ownership?
+    compilerError("A bytes cannot be passed to createBytesWithOwnedBuffer");
+  }
+
+  /*
+    Creates a new bytes which takes ownership of the internal buffer of a
+    `c_string`.
```

Comment:
I'd elaborate on what "takes ownership" means, e.g. adding this sentence here:

> The buffer will be freed when the `bytes` is deinitialized.

Please apply this change to the other createXWithOwnedBuffer functions.